packages/labs/src/components/omnibox/omnibox.tsx

```diff
@@ -0,0 +1,197 @@
+    private queryList: QueryList<T>;
+    private refHandlers = {
+        queryList: (ref: QueryList<T>) => this.queryList = ref,
+    };
+
+    public render() {
+        // omit props specific to this component, spread the rest.
+        const { isOpen, itemRenderer, inputProps, noResults, overlayProps, ...props } = this.props;
+        return <this.TypedQueryList
+            {...props}
+            activeItem={this.state.activeItem}
+            onActiveItemChange={this.handleActiveItemChange}
+            onItemSelect={this.handleItemSelect}
+            query={this.state.query}
+            ref={this.refHandlers.queryList}
+            renderer={this.renderQueryList}
+        />;
+    }
+
+    public componentWillReceiveProps(nextProps: IOmniboxProps<T>) {
+        const { isOpen } = nextProps;
+        const canClearQuery = !this.props.isOpen && isOpen && this.props.resetOnSelect;
+
+        this.setState({
+            activeItem: canClearQuery ? this.props.items[0] : this.state.activeItem,
+            query: canClearQuery ? "" : this.state.query,
+        });
+    }
+
+    private renderQueryList = (listProps: IQueryListRendererProps<T>) => {
+        const { inputProps = {}, isOpen, overlayProps = {} } = this.props;
+        const { query } = this.state;
+        const { ref, ...htmlInputProps } = inputProps;
+        const { handleKeyDown, handleKeyUp } = listProps;
+
+        return (
+            <Overlay
+                hasBackdrop={true}
+                {...overlayProps}
+                isOpen={isOpen}
+                className={classNames(overlayProps.className, `${Classes.OMNIBOX}-overlay`)}
+                onClose={this.handleOverlayClose}
+            >
+                <div
+                    className={classNames(listProps.className, Classes.OMNIBOX)}
+                    onKeyDown={isOpen && query.length > 0 && handleKeyDown}
+                    onKeyUp={isOpen && query.length > 0 && handleKeyUp}
+                >
+                    <InputGroup
+                        autoFocus={true}
+                        className={CoreClasses.LARGE}
+                        leftIconName="search"
+                        onChange={this.handleQueryChange}
+                        placeholder="Search..."
+                        value={listProps.query}
+                        {...htmlInputProps}
+                    />
+                    {this.maybeRenderMenu(listProps)}
+                </div>
+            </Overlay>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IQueryListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        if (filteredItems.length === 0) {
+            return noResults;
+        }
+        return filteredItems.map((item, index) => itemRenderer({
+            index,
+            item,
+            handleClick: (e) => handleItemSelect(item, e),
+            isActive: item === activeItem,
```

Comment:
`===` is def not sufficient, this has come up in chat a few times. 
easy way to break it is to create new objects in each render (rather than reusing existing ones), which is a very common JS pattern.

we'll need to add an `isEqual` prop with a default `===` implementation so folks can define their own equality checks.